File: linked-list/recorrerLista/recorrerLista.ts
// Importamos la definición de la lista enlazada desde "../type"
import { LinkedList } from "../type";

// Función para recorrer una lista enlazada e imprimir sus datos
export const recorrerLista = (l: LinkedList) => {
  while (l.head !== null) {
    // hacer algo con los datos
    console.log(l.head.data); // Imprimimos el valor del campo 'data' del nodo actual
    l.head = l.head.next; // Avanzamos al siguiente nodo
  }
}

File: linked-list/recorrerLista/recorrerLista.pas
// Procedimiento para recorrer una lista enlazada e imprimir sus datos
procedure recorrerLista(l: lista);
begin
    while (l <> NIL) do // Mientras no lleguemos al final de la lista
    begin
        { hacer algo con los datos } // Comentario: Aquí puedes realizar alguna operación con los datos del nodo
        writeln(l^.dato); // Imprimimos el valor del campo 'dato' del nodo actual
        l := l^.sig; // Avanzamos al siguiente nodo
    end;
end;
